### PR TITLE
[vcpkg-find-acquire-program] Revise test port

### PR DESCRIPTION
**Pull request by dg0yt**
Complements #30780.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The "supports" clause reflects platforms that may be fixed by this new version

### CODE REVIEW

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. Please convert PR to ready for review after fixing CI errors:
```
CMake Error at scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake:48 (message):
  The following programs do not exist:

     BAZEL
     BISON
     FLEX
     GIT
     GN
     MESON
     NINJA
     PERL
     PKGCONFIG
     PYTHON2
     PYTHON3
     SCONS
     SWIG
     YASM
     7Z
     ARIA2
     CLANG
     DARK
     DOXYGEN
     GASPREPROCESSOR
     GO
     GPERF
     JOM
     NASM
     NUGET
     RUBY
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)
```

**Comment by dg0yt** [comment]
I already knew that the old test was a no-op partially... It didn't process the list of programs which was meant to be tested for existence. Now l learned that it would have failed for programs which need interpreters:
~~~
  The following programs do not exist:

     MESON: D:/downloads/tools/python/python-3.10.7-x64/python.exe
     D:/downloads/tools/meson/meson-aeda7f249c4a5dbbecc52e44f382246a2377b5b0/meson.py
     GASPREPROCESSOR: D:/downloads/tools/perl/5.32.1.1/perl/bin/perl.exe
     D:/downloads/tools/gas-preprocessor/9309c67a/gas-preprocessor.pl
     SCONS: D:/downloads/tools/python/python-2.7.18-x64/python.exe
     D:/downloads/tools/scons/4.1.0/scons.py
~~~

**Comment by dg0yt** [comment]
It is not just https://github.com/microsoft/vcpkg/labels/category%3Acode-cleanup: It actually ensures proper ABI hashing, and thus proper triggering of the CI port. Without spelling out `vcpkg_find_acquire_program(ACTUAL_NAME)` in the portfile, PR CI may fail to detect problems in PRs changig `vcpkg_find_acquire_program(ACTUAL_NAME).cmake`.

**Comment by dg0yt** [comment]
After some confusion, programs with interpreters are now handled properly (again).
~~~
  The following programs do not exist:

     EXPECTED_FAIL: /PROGRAM
     EXPECTED_FAIL (interpreter): /INTERPRETER
~~~

**Review by Adela0814 — DISMISSED** [review]

File: scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake [hunk 1]
```diff
@@ -1,21 +1,62 @@
...
+    vcpkg_find_acquire_program(RUBY)
+    vcpkg_find_acquire_program(SCONS)
+    vcpkg_find_acquire_program(SWIG)
+endif()
+
+set(missing "")
+foreach(variable IN LISTS variables)
+    list(POP_BACK "${variable}" program)
+    if(NOT EXISTS "${program}")
+        list(APPEND missing "${program}: ${${program}}")
```
Comment (dan-shaw) [inline scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake:52]: Shouldn't this be 
```suggestion
        list(APPEND missing "${variable}: ${program}")
```

Comment (dg0yt) [reply scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake]: Good catch. The last commit reverted too much. 
I pushed a fix.

**Pull request merged**